src/apiProxy/api-proxy.js

```diff
@@ -27,20 +37,28 @@ import { duotoneRef } from '../util/duotone';
  * @param response {String} the raw response body text from an API call
  * @return responseObj the JSON-parsed text, possibly with error info
  */
-export const parseApiResponse = response => {
-	let responseObj;
-	let error;
+export const parseApiResponse = ([response, body]) => {
+	let value;
+	const flags = parseResponseFlags(response);
 
-	try {
-		responseObj = JSON.parse(response);
-	} catch(e) {
-		error = `API response was not JSON: "${response}"`;
+	if (response.statusCode > 299) {
```

Comment:
nitpick, but i'd almost just rather check if the statusCode doesn't start with '2' instead, unless you're handling `1xx` informational codes as a success.